src/web/app/pages-admin/admin-home-page/admin-home-page.component.ts

```diff
@@ -85,29 +85,36 @@ export class AdminHomePageComponent {
    * Adds the instructor at the i-th index.
    */
   addInstructor(i: number): void {
+    this.asyncAddInstructor(i).then();
+  }
+
+  /**
+   * Adds the instructor at the i-th index.
+   */
+  async asyncAddInstructor(i: number): Promise<JoinLink | null> {
     const instructor: InstructorData = this.instructorsConsolidated[i];
     if (instructor.status !== 'PENDING' && instructor.status !== 'FAIL') {
-      return;
+      return null;
     }
     this.activeRequests += 1;
     instructor.status = 'ADDING';
 
     this.isAddingInstructors = true;
-    this.accountService.createAccount({
+    return this.accountService.createAccount({
       instructorEmail: instructor.email,
       instructorName: instructor.name,
       instructorInstitution: instructor.institution,
-    })
-        .pipe(finalize(() => this.isAddingInstructors = false))
-        .subscribe((resp: JoinLink) => {
-          instructor.status = 'SUCCESS';
-          instructor.joinLink = resp.joinLink;
-          this.activeRequests -= 1;
-        }, (resp: ErrorMessageOutput) => {
-          instructor.status = 'FAIL';
-          instructor.message = resp.error.message;
-          this.activeRequests -= 1;
-        });
+    }).pipe(finalize(() => this.isAddingInstructors = false))
```

Comment:
Probably don't need two pipe, you can just place the `tap` into the first pipe method